src/zcash/NoteEncryption.hpp

```diff
@@ -61,13 +61,18 @@ class NoteDecryption {
     typedef boost::array<unsigned char, CLEN> Ciphertext;
     typedef boost::array<unsigned char, MLEN> Plaintext;
 
+    // Unused default constructor to make allocators happy
+    NoteDecryption() { }
     NoteDecryption(uint256 sk_enc);
 
     Plaintext decrypt(const Ciphertext &ciphertext,
                       const uint256 &epk,
                       const uint256 &hSig,
                       unsigned char nonce
                      ) const;
+
+    friend inline bool operator==(const NoteDecryption& a, const NoteDecryption& b) { return a.sk_enc == b.sk_enc && a.pk_enc == b.pk_enc; }
+    friend inline bool operator<(const NoteDecryption& a, const NoteDecryption& b) { return a.pk_enc < b.pk_enc; }
```

Comment:
All these operator defines (here and elsewhere in the PR) were added simply to make `std::map` happy (which keeps elements ordered internally). Technically, the ordering doesn't matter for us; practically, it may as well be valid, so I'll tweak this to compare both fields.